modules/ding_place2book/ding_place2book.module

```diff
@@ -442,61 +442,68 @@ function ding_place2book_get_event_state($event_maker_id, $event_id) {
     $p2b = ding_place2book_instance();
     $event = $p2b->getEvent($event_maker_id, $event_id);
     $meta_data = $event->meta_data;
+    $event_state = $meta_data->event_state;
     $start = strtotime($event->begin_at);
 
-    if (time() < $start) {
-      // Go through the event state.
-      switch ($meta_data->event_state) {
-        case 'published':
-          // Check if waiting list is activated.
-          switch ($meta_data->waiting_list) {
-            case 'true':
-              $status = 'waiting-list';
-              break;
-
-            default:
-              // Go through the sales status.
-              switch ($meta_data->sales_status) {
-                case 'open':
-                  // Check if there is any available tickets for sale.
-                  if ($meta_data->available_tickets > 0 || strval($meta_data->available_tickets) == '∞') {
-                    $status = 'open';
-                  }
-                  else {
-                    $status = 'sold-out';
-                  }
-                  break;
-
-                default:
-                  $status = 'closed';
-                  break;
-              }
-              break;
+    // We start out with the closed status and look in the event data to prove
+    // us wrong.
+    $status = 'closed';
+
+    // First we check the event_state which can be 'published', 'draft',
+    // 'passive', 'held', 'pause' or 'completed'.
+    // If the event is on 'pause' or is a 'draft' we will translate that to our
+    // 'closed' event. We don't know if these event will ever be unpaused or
+    // undrafted, so it doesn't make sense to show anything other than closed.
+    // If the event is 'held' or 'completed' we will just show event-over to the
+    // users. For them the difference between these two is irelevant.
+    // If the event is 'passive' we won't show anything at all in frontend.
+    // If the event is 'published' we will look at available tickets and their
+    // sales dates to determine the proper ding_event event state, which in this
+    // case can be either 'waiting-list', 'open', 'sold-out' or 'closed'.
+    if (time() < $start && $event_state === 'published') {
```

Comment:
Separate these statements and comment on them. It eases future development and makes diffs better.

Reply:
You want me to separate `time() < $start` and `$event_state === 'published'` into different if-statements, where the last one is nested?